api/pkg/hystrix/client.go

```diff
@@ -0,0 +1,47 @@
+package hystrix
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/afex/hystrix-go/hystrix"
+)
+
+type Doer interface {
+	Do(request *http.Request) (*http.Response, error)
+}
+
+type Client struct {
+	client      Doer
+	commandName string
+}
+
+func NewClient(client Doer, circuitConfig *hystrix.CommandConfig, commandName string) *Client {
+	config := &hystrix.CommandConfig{}
+	if circuitConfig != nil {
+		config = circuitConfig
+	}
+
+	hystrix.ConfigureCommand(commandName, *config)
+	return &Client{
+		client:      client,
+		commandName: commandName,
+	}
+}
+
+func (cl *Client) Do(request *http.Request) (*http.Response, error) {
+	var response *http.Response
+	var err error
+
+	err = hystrix.Do(cl.commandName, func() error {
+		response, err = cl.client.Do(request)
+		if err != nil {
+			return err
+		}
+		if response.StatusCode >= http.StatusInternalServerError {
+			return errors.New("got 5xx response code")
```

Comment:
Can you add response code the the error message

Reply:
Also is there a way if we can also read the response payload and add error logging here.

Reply:
I think for response payload can be read by the caller instead of in the client itself